classes/class-wc-rest-connect-migration-flag-controller.php

```diff
@@ -31,6 +56,7 @@ public function post( $request ) {
 		$result = WC_Connect_Options::update_option( 'wcshipping_migration_state', $migration_state );
 
 		if ( $result ) {
+			$this->tracks->record_user_event( 'migration_flag_updated', array( 'migration_state' => $migration_state ) );
```

Comment:
Do you think it's fine to only track it when the migration state has updated? what's against always tracking when this endpoint is called? Wouldn't that give us more insight on what's happening on the merchant sites when it gets to migration?

Reply:
we can also track when errors are triggered, but I was thinking about this metrics more like product metrics, instead of debugging info. The errors defined in that class are not supposed to happen: we should use a different tool (ie. sentry? or whatever we use to catch errors) and fix them, but I don't think we should reflect errors in the product metrics. Do you think we should track them too? I'm ok with implementing that, anyway, if you think it would be valuable :+1:

Reply:
I agree that other tools could be a better fit for tracking none-updating requests, however we don't use them at this point so and tracking too much is always better than tracking too less, how about adding the update value ($result) to the tracked data, something like the following:

```
$this->tracks->record_user_event( 'migration_flag_state_update', array( 'migration_state' => $migration_state, 'updated => $result ) );
```
I'd leave it to you to decide if you want to change it though.

Reply:
that's a great idea, I'll add it to the tracking data :+1: